standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/entitymappers/SqlParamSourceMapper.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore.txn.entitymappers;
+
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+
+public interface SqlParamSourceMapper<T> {
```

Comment:
What is the intention of that interface?

Reply:
It's the counterpart of the `RowMapper` interface, map an entity to `SqlParamSource`. Entity mappers can implement one of them or both. Currently it is not in use, but can be handy in the entity mappers need to be handled together for any reason.